jina/orchestrate/deployments/__init__.py

```diff
@@ -742,13 +742,28 @@ def _parse_devices(value: str, num_devices: int):
                 parts = value.split(':')
 
                 if len(parts) == 1:
-                    # slice(stop)
+                    try:
+                        int(parts[0])
+                    except:
+                        use_uuids = True
+                    if use_uuids:
+                        return parts
                     parts = [parts[0], str(int(parts[0]) + 1)]
-                # else: slice(start, stop[, step])
             else:
-                return [int(p) for p in parts]
+                # try to detect if parts are not numbers
+                try:
+                    int(parts[0])
+                except:
+                    use_uuids = True
+
+                if not use_uuids:
+                    return [int(p) for p in parts]
+                else:
+                    return parts
```

Comment:
Is it supported to have only some parts as UUID? Should we do this check for every part?

Reply:
I understand that if u pass the first as UUID, all of them will, other checks feel like overkill